### PR TITLE
devices: power: add IP9820 power switch section

### PR DESCRIPTION
**Pull request by SyHuangCBN**
Due to we still have IP9820 power swith to use.
We need to add power switch section.

Signed-off-by: sy_huang <sy_huang@compalbn.com>

### CODE REVIEW

**Comment by SyHuangCBN** [comment]
I saw it shows failure, not sure what's failure with IP9820. Will keep tracking.

File: devices/power.py [hunk 1]
```diff
@@ -64,6 +64,8 @@ def get_power_device(ip_address, username=None, password=None, outlet=None):
         return Ip9258(ip_address, outlet, username=username, password=password)
     if 'Cyber Power Systems' in data:
         return CyberPowerPdu(ip_address, outlet=outlet, username=username, password=password)
+    if 'Unauthorized' in data:
+        return Ip9820(ip_address, outlet)
```
Comment (mattsm) [inline devices/power.py]: Why Unauthorized? That might match other http data in other devices. Can you please share the full output?

File: devices/power.py [hunk 2]
```diff
@@ -330,6 +332,45 @@ def reset(self):
...
+        return urllib2.urlopen('http://' + self._ip_address + '/set.cmd?cmd=setpower+p6' + str(self.port) + '=0')
+
+    def reset(self):
+        self.off()
+        time.sleep(5)
+        self.on()
+
+    def getPowerStatus(self):
+
+        oid = self.oid_Outlet + '.' + str(self.port) + '.0'
```
Comment (mattsm) [inline devices/power.py]: Where does self.oid_Outlet get set? Please just remove these functions if they were used for testing.

File: devices/power.py (hunk 1 above)
Comment (lynnlincbn) [reply devices/power.py]: We can't get power device name to mask sure which power device in use.
Only can get below message, so we use 'Unauthorized' to check power device is Ip9820.

Please refer below http data message:

<html><head><title>Document Error: Unauthorized</title></head>
--
  | <body><h2>Access Error: Unauthorized</h2>
  | <p>Access to this document requires a User ID</p></body></html>

File: devices/power.py (hunk 2 above)
Comment (lynnlincbn) [reply devices/power.py]: Will remove it, thanks.

File: devices/power.py (hunk 1 above)
Comment (mattsm) [reply devices/power.py]: I understand that, but I think you need to try to login and see what model this is... if you just do Unauthorized you will match ANY device that has that text in their output. Well that's more than just this device. You need to think generically.

I would say, if you don't match anything without logging in- then you do just that login and see if you can find a model number in the output and match off that.

**Comment by mattsm** [comment]
Also, please rebase this change on the latest master... we've since fixed the travis ci issues.

**Comment by SyHuangCBN** [comment]
Update #57 
Closed it.

**Pull request closed without merging**